tools/serverpod_cli/lib/src/analyzer/dart/endpoint_analyzers/endpoint_method_analyzer.dart

```diff
@@ -90,7 +90,12 @@ abstract class EndpointMethodAnalyzer {
 
   static bool _missingSessionParameter(List<ParameterElement> parameters) {
     if (parameters.isEmpty) return true;
-    return parameters.first.type.element?.displayName != 'Session';
+
+    bool firstParameterIsNotSession =
+        parameters.first.type.element?.displayName != 'Session';
+    if (firstParameterIsNotSession) return true;
+
+    return parameters.first.isNamed;
```

Comment:
Question: What happens if the session parameter is Optional? That is:
```dart
Future<int> myMethod([Session session]);
```

Maybe we should add a test case for this and possibly modify the logic here to handle that case?

Reply:
Nice catch!

Reply:
Indeed this also cause the bug (Fixed in [b8e60b5](https://github.com/serverpod/serverpod/pull/2597/commits/b8e60b57c254ba84ba88de20fd2912a4de0eea04))💡. As a follow-up question then, the following is currently interpreted as a valid endpoint definition (note the `?`). Technically it will always work, but I wonder if this should result in an `info` or `warning` message from the generator script?
```dart
  Future<void> hello(Session? session) async {
    return null;
  }
```